Source/DiabloUI/selgame.cpp

```diff
@@ -282,40 +282,40 @@ void selgame_GameSelection_Select(int value)
 	switch (value) {
 	case 0:
 	case 1: {
-		title = _("Create Game");
+		title = _("Create Game").c_str();
 
 		SDL_Rect rect4 = { (Sint16)(PANEL_LEFT + 299), (Sint16)(UI_OFFSET_Y + 211), 295, 35 };
-		vecSelGameDialog.push_back(std::make_unique<UiArtText>(_("Select Difficulty"), rect4, UiFlags::AlignCenter | UiFlags::FontSize30 | UiFlags::ColorUiSilver, 3));
+		vecSelGameDialog.push_back(std::make_unique<UiArtText>(_("Select Difficulty").c_str(), rect4, UiFlags::AlignCenter | UiFlags::FontSize30 | UiFlags::ColorUiSilver, 3));
 
-		vecSelGameDlgItems.push_back(std::make_unique<UiListItem>(_("Normal"), DIFF_NORMAL));
-		vecSelGameDlgItems.push_back(std::make_unique<UiListItem>(_("Nightmare"), DIFF_NIGHTMARE));
-		vecSelGameDlgItems.push_back(std::make_unique<UiListItem>(_("Hell"), DIFF_HELL));
+		vecSelGameDlgItems.push_back(std::make_unique<UiListItem>(_("Normal").c_str(), DIFF_NORMAL));
+		vecSelGameDlgItems.push_back(std::make_unique<UiListItem>(_("Nightmare").c_str(), DIFF_NIGHTMARE));
+		vecSelGameDlgItems.push_back(std::make_unique<UiListItem>(_("Hell").c_str(), DIFF_HELL));
 
 		vecSelGameDialog.push_back(std::make_unique<UiList>(vecSelGameDlgItems, vecSelGameDlgItems.size(), PANEL_LEFT + 300, (UI_OFFSET_Y + 282), 295, 26, UiFlags::AlignCenter | UiFlags::FontSize24 | UiFlags::ColorUiGold));
 
 		SDL_Rect rect5 = { (Sint16)(PANEL_LEFT + 299), (Sint16)(UI_OFFSET_Y + 427), 140, 35 };
-		vecSelGameDialog.push_back(std::make_unique<UiArtTextButton>(_("OK"), &UiFocusNavigationSelect, rect5, UiFlags::AlignCenter | UiFlags::VerticalCenter | UiFlags::FontSize30 | UiFlags::ColorUiGold));
+		vecSelGameDialog.push_back(std::make_unique<UiArtTextButton>(_("OK").c_str(), &UiFocusNavigationSelect, rect5, UiFlags::AlignCenter | UiFlags::VerticalCenter | UiFlags::FontSize30 | UiFlags::ColorUiGold));
 
 		SDL_Rect rect6 = { (Sint16)(PANEL_LEFT + 449), (Sint16)(UI_OFFSET_Y + 427), 140, 35 };
-		vecSelGameDialog.push_back(std::make_unique<UiArtTextButton>(_("CANCEL"), &UiFocusNavigationEsc, rect6, UiFlags::AlignCenter | UiFlags::VerticalCenter | UiFlags::FontSize30 | UiFlags::ColorUiGold));
+		vecSelGameDialog.push_back(std::make_unique<UiArtTextButton>(_("CANCEL").c_str(), &UiFocusNavigationEsc, rect6, UiFlags::AlignCenter | UiFlags::VerticalCenter | UiFlags::FontSize30 | UiFlags::ColorUiGold));
 
 		UiInitList(selgame_Diff_Focus, selgame_Diff_Select, selgame_Diff_Esc, vecSelGameDialog, true);
 		break;
 	}
 	case 2: {
-		title = _("Join TCP Games");
+		title = _("Join TCP Games").c_str();

```

Comment:
Should we change `title` to `string_view`?

Reply:
That would require for `UiArtText` to accept it instead of `const char *` which is a bit complicated. I thought it might have been done in https://github.com/diasurgical/devilutionX/pull/4196 but looks like that isn't the case. I updated this PR to also handle the other UI classes except `UiArtText` and `UiEdit`. I think they should be tackled in a follow up.